### PR TITLE
Add support for RMI

### PR DESCRIPTION
**Pull request by ftomassetti**
By making elements Serializable we can expose the parsers through RMI, as we do in the Audit Viewer

### CODE REVIEW

File: core/src/main/kotlin/com/strumenta/kolasu/model/Model.kt [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 
+import java.io.Serializable
 import kotlin.reflect.KClass
 import kotlin.reflect.KProperty1
 import kotlin.reflect.KVisibility
 import kotlin.reflect.full.findAnnotation
 import kotlin.reflect.full.memberProperties
 
-interface Origin {
+interface Origin : Serializable {
```
Comment (alessiostalla) [inline core/src/main/kotlin/com/strumenta/kolasu/model/Model.kt:10]: Is this necessary? Not all Origins will be serializable...

File: core/src/main/kotlin/com/strumenta/kolasu/model/Model.kt [hunk 2]
```diff
@@ -21,7 +22,7 @@ data class CompositeOrigin(
     override val sourceText: String?
 ) : Origin
 
-interface Destination
+interface Destination : Serializable
```
Comment (alessiostalla) [inline core/src/main/kotlin/com/strumenta/kolasu/model/Model.kt:25]: Same here

**Review by alessiostalla — COMMENTED** [review]
In general, I see no problem in making core data classes (e.g. Position) serializable.
With nodes, I'm a bit afraid of future compatibility problems. If we later implement AST nodes on top of LionWeb nodes, are we sure that everything in LionWeb is serializable, and that it doesn't drag all of the metamodels into the object graph?

File: core/src/main/kotlin/com/strumenta/kolasu/model/Model.kt (hunk 1 above)
Comment (ftomassetti) [reply core/src/main/kotlin/com/strumenta/kolasu/model/Model.kt:10]: Yes, but a Node could specify an Origin or a Destination, so all Origins and all Destinations should be Serializable

Comment (alessiostalla) [reply core/src/main/kotlin/com/strumenta/kolasu/model/Model.kt:10]: ParseTreeOrigin isn't, though, for example. IMHO it's more appropriate if only actual serializable concrete classes implement Serializable.

**Comment by ftomassetti** [comment]
> In general, I see no problem in making core data classes (e.g. Position) serializable. With nodes, I'm a bit afraid of future compatibility problems. If we later implement AST nodes on top of LionWeb nodes, are we sure that everything in LionWeb is serializable, and that it doesn't drag all of the metamodels into the object graph?

This is a good question. I think that if our nodes extends the lionweb nodes and we send our nodes only and our nodes are serializable then it should work. For the metamodel classes... yes, they may also need to be serializable. Perhaps I could do that change on the LionWeb side, when needed. It is just about implementing a marker interface, so it should not cause any specific problem. We could do that if at that stage RMI is not presenting problems and we want to confirm we want to use it

File: core/src/main/kotlin/com/strumenta/kolasu/model/Model.kt (hunk 1 above)
Comment (ftomassetti) [reply core/src/main/kotlin/com/strumenta/kolasu/model/Model.kt]: You are right, I make only specific subclasses Serializable now

**Comment by ftomassetti** [comment]
@alessiostalla I addressed the comments. Would this PR now be acceptable?

File: core/src/main/kotlin/com/strumenta/kolasu/model/Model.kt (hunk 2 above)
Comment (ftomassetti) [reply core/src/main/kotlin/com/strumenta/kolasu/model/Model.kt]: Changed

**Review by alessiostalla — APPROVED** [review]

**Pull request merged**